### PR TITLE
Message compose

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @danehansen @dwick

### CODE REVIEW

File: src/views/components/Modal.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Modal extends BaseComponent {
     }
   }
 
-  componentWillRecieveProps (nextProps) {
+  componentWillReceiveProps (nextProps) {
```
Comment (dwick) [inline src/views/components/Modal.jsx]: separate commit for this

File: src/views/pages/messageCompose.jsx [hunk 2]
```diff
@@ -1,24 +1,273 @@
...
+
+  sendMessage (e) {
+    e.preventDefault();
+    // etc etc
+
+    let text = this.state.text;
+    let recipient = this.state.recipient;
+    let subject = this.state.subject;
+    // TODO: let from_sr = this.refs.subject.getDomNode().value;
+    var errors = this._validateFields(recipient, subject, text);
```
Comment (dwick) [inline src/views/pages/messageCompose.jsx]: i would think this logic should belong to the `Message` model

Comment (curioussavage) [inline src/views/pages/messageCompose.jsx]: I'm inclined to leave this separate for now. The validate function on the model is just meant to return a bool, this method is just there so I can highlight fields that need to be filled. needs to return the names of the fields. bad naming maybe?

**Comment by curioussavage** [comment]
:haircut:

File: src/views/pages/messageCompose.jsx (hunk 2 above)
Comment (dwick) [inline src/views/pages/messageCompose.jsx]: looks to me as if it returns `true` if valid or a list of invalid fields if invalid: https://github.com/reddit/snoode/blob/31595436ac25e4fe758d3a149b6a593e8ae13e7e/src/models/base.es6.js#L66

**Comment by dwick** [comment]
:nail_care:

File: src/views/pages/messageCompose.jsx (hunk 2 above)
Comment (curioussavage) [inline src/views/pages/messageCompose.jsx]: oops. thanks for pointing that out. will do then

**Comment by curioussavage** [comment]
:eyeglasses: @ajacksified 

this one is ready now

**Comment by ajacksified** [comment]
Needs some design changes to bring it in line with submit; :nail_care:

**Comment by curioussavage** [comment]
messagecompose-change
messagecompose-old

**Comment by curioussavage** [comment]
@librarianavenger @tdohz   want some feedback on this form. Never had any design attention so I just built it to look like the submit page. 

The bottom is how it looks like right now, with the exception that my snooboots theme has dark buttons (need to fix that). 

The image above is another possible layout (not styled well right now) with the from select box on the bottom. Can I get some input on this?

I would really like to get this out, the branch has been sitting here done for a while now. Design tweaks are only holdup

**Comment by dwick** [comment]
:running:

**Comment by curioussavage** [comment]
I restyled this view per advice from jack last week. Also refactored the css to fit with new style
messagecompose

**Comment by curioussavage** [comment]
:haircut:

**Comment by schwers** [comment]
note: the select for username / subreddits shouldn't be a select if the user isn't moderating any subredits

File: assets/less/components/messageCompose.less [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    &:focus {
+      outline: none;
+    }
+  }
+
+  &__from-wrap {
+    padding-left: 10px;
+    padding-top: 15px;
+    display: inline-block;
+    width: calc(~'100% - 170px');
```
Comment (schwers) [inline assets/less/components/messageCompose.less]: double check layout is okay-ish when calc isn't supported (eg android 4.3)

**Comment by curioussavage** [comment]
closing this as it has sat here for ages and niranjan is working on a new messages ui

**Pull request closed without merging**